tests/test_latent_diffusion.py

```diff
@@ -0,0 +1,343 @@
+import os
+import time
+from datetime import datetime
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import yaml
+from diffusers import DDIMScheduler, DDPMPipeline, StableDiffusionPipeline
+from diffusers.models import AutoencoderKL
+from loguru import logger
+from torch.utils.data import DataLoader
+from torchvision import transforms
+from tqdm import tqdm
+
+import wandb
+from xfetus.utils.datasets import (AfricanFetalPlaneDataset,
+                                   PrecomputedFetalPlaneDataset)
+
+file=Path().absolute()/"tests/config_test_latent_diffusion.yml"
+with open(file, "r") as file:
+   config_yaml = yaml.load(file, Loader=yaml.FullLoader)
+
+
+def test_data_path():
+   """
+   Test data path
+   pytest -vs tests/test_latent_diffusion.py::test_data_path
+   TODO:
+   https://github.com/ashleve/lightning-hydra-template/blob/main/tests/conftest.py
+   """
+   MODELS_PATH = os.path.join(str(Path.home()), config_yaml["MODELS_PATH"])
+   DATASET_PATH = os.path.join(str(Path.home()), config_yaml["SPANISH_FETAL_PLANES_DATA_PATH"])
+   batch_size = config_yaml["model_hyperparameters"]["batch_size"]
+
+   # define filenames for training data (saved as several numpy arrays)
+   training_filenames = [
+      'Fetal_abdomen_train.npy',
+      'Fetal_brain_train.npy',
+      'Fetal_femur_train.npy',
+      'Fetal_thorax_train.npy',
+      'Maternal_cervix_train.npy',
+      'Other_train.npy',
+   ]
+
+   # define filenames for validation data (saved as several numpy arrays)
+   validation_filenames = [
+      'Fetal_abdomen_validation.npy',
+      'Fetal_brain_validation.npy',
+      'Fetal_femur_validation.npy',
+      'Fetal_thorax_validation.npy',
+      'Maternal_cervix_validation.npy',
+      'Other_validation.npy',
+   ]
+
+   # Define augmentations for each image
+   transform_operations = transforms.Compose([
+      transforms.ToTensor(),
+      transforms.RandomHorizontalFlip(p=0.5),
+      transforms.RandomRotation(45),
```

Comment:
We should maybe reconsider what augmentations we are doing here. But these are fine for now as a placeholder.